azure-quantum/azure/quantum/_authentication/_default.py

```diff
@@ -97,7 +99,18 @@ def _initialize_credentials(self):
             credentials.append(VisualStudioCodeCredential(authority=self.authority, tenant_id=self.tenant_id))
             credentials.append(AzureCliCredential(tenant_id=self.tenant_id))
             credentials.append(AzurePowerShellCredential(tenant_id=self.tenant_id))
-            credentials.append(InteractiveBrowserCredential(authority=self.authority, tenant_id=self.tenant_id))
+            # Before trying other credential types, try to use already cached token.
+            credentials.append(SharedTokenCacheCredential(authority=self.authority))
+            credentials.append(
+                InteractiveBrowserCredential(
+                    authority=self.authority, 
+                    tenant_id=self.tenant_id,
+                    cache_persistence_options=TokenCachePersistenceOptions(
+                        # Not all linux systems has preinstalled libsecret, which is required for storage encryption. 
+                        allow_unencrypted_storage=True
```

Comment:
In order to make it work on Linuux, the  `azure.identity` library relies on `libsecret` which may not be installed. 
The quick workaround would be to use unencrypted caching, but I am pretty sure that's not a good idea. 
As better option, we can pre-check for the `libsecret` on linux platform by ourselves, so it would be narrowed down to that platform.

Reply:
What happens if `libsecret` was not found? Would it throw an exception? If it would, it might be fine, because the `_ChainedTokenCredential` will catch it and will try the `InteractiveBrowserCredential` next.

Reply:
If `libsecret` is not found `SharedTokenCacheCredential` throws `CredentialUnavailableError` which will be caught by `_ChainedTokenCredential` as you mentioned.